gitlab-runner/src/client.rs

```diff
@@ -248,6 +251,18 @@ impl Client {
         }
     }
 
+    fn check_for_job_cancellation(&self, response: &reqwest::Response) -> Result<(), Error> {
+        let job_status = response
+            .headers()
+            .get(JOB_STATUS)
+            .and_then(|v| v.to_str().ok());
+        if let Some("canceled") = job_status {
+            Err(Error::JobCancelled)
+        } else {
+            Ok(())
+        }
+    }
+
```

Comment:
```suggestion
    fn check_for_job_cancellation(&self, response: &reqwest::Response) -> Result<(), Error> {
       match response.headers().get(JOB_STATUS) {
           Some(h) if h == "canceled" => Err(Error::JobCancelled),                            
           _ => Ok(()),                                                                                                                             
        }    
    }
```
seems quite a bit more elegant ? :)

Reply:
Nice, I didn't realize `HeaderValue` had `PartialEq<str>`.